unmarshal_test.go

```diff
@@ -314,6 +384,16 @@ func TestUnmarshal(t *testing.T) {
 			},
 			expect:      &articleRelatedCommentsNested,
 			expectError: nil,
+		}, {
+			description: "links member only",
```

Comment:
Managed to rewrite the solution without an additional `json.Unmarshal` but at the expense of this undesired behavior change.

How it normally works now: https://go.dev/play/p/eeC-72rSJeN

Reply:
Resolved by partial unmarshaling